# Task 03 - Business Intelligence Dashboard

### 1. Business Understanding

### 1.1 Problem Statement

The dataset contains transactional sales data that captures the relationship between customers, products, and purchasing activities. The primary challenge is to understand how customer characteristics, product categories, locations, and purchasing patterns influence overall sales performance. By analyzing these interactions, the business can identify meaningful patterns, monitor sales performance, and support data-driven decision-making for future business strategies.

### 1.2 About the Dataset
This dataset represents retail sales transactions involving customers, products, and orders. It captures how different customers purchase different products over time and records the resulting sales generated from those transactions.

The dataset combines customer information, product details, order information, sales values, and geographical information, making it suitable for business intelligence, customer analytics, sales analysis, and dashboard development.

#### Unit of Analysis

The unit of analysis in this dataset is a transaction line item.

Each row represents a single product purchased as part of a customer order rather than an entire customer or a complete order.

A single Order ID may appear multiple times because one order can contain multiple products. Therefore, Order ID should be treated differently from the number of rows while performing business analysis.

#### Main Business Entities

The dataset mainly consists of three business entities:

• Customer
    - Identified using Customer ID
    - Contains demographic information such as Age, Gender and City.

• Order
    - Identified using Order ID and Order Date.
    - Represents purchasing activity performed by customers.

• Product
    - Includes Product Name, Category and Sales information.
    - Represents items purchased during customer transactions.

#### Types of Analysis Supported

This dataset supports multiple business analyses, including:

- Customer Purchasing Behaviour Analysis
- Customer Segmentation Analysis
- Product and Category Performance Analysis
- Time-based Sales Trend Analysis
- Geographical Sales Performance Analysis

#### Dataset Limitations

The dataset does not contain some important business information, including:

- Profit or Cost
- Store or Branch information
- Online/Offline purchase channel
- Customer Reviews or Ratings
- Referral Source
- Marketing Campaign data
- Customer Satisfaction or Returns

Therefore, profitability analysis, customer satisfaction analysis, and marketing effectiveness analysis cannot be performed directly using this dataset.

### 3.3 Business Objectives

1. Understand customer purchasing behavior across different customer segments.

2. Evaluate sales performance across products, categories, cities, and time periods.

3. Identify business opportunities and high-performing areas using sales data.

4. Support data-driven business decision-making through analytical insights and interactive dashboards.

### 1.4 Stakeholders

The primary stakeholders for this analysis are:

#### 1. Business Owner / CEO
- Monitor overall business performance.
- Track sales growth and customer purchasing trends.
- Make strategic business decisions based on business insights.

#### 2. Sales Manager
- Monitor sales performance across products and categories.
- Identify high-performing and low-performing products.
- Improve sales strategies using performance trends.

#### 3. Marketing Manager
- Identify target customer segments.
- Design customer-focused marketing campaigns.
- Promote products based on customer purchasing behaviour and seasonal demand.




### 1.5 Business Questions

#### Customer Purchasing Behaviour

- Which age group generates the highest sales?
- Which age group has the highest average sales per customer?
- Which gender contributes more to overall sales?
- Do loyal customers contribute significantly to total revenue?
- What is the average order value for each customer segment?

---

#### Product Performance

- Which products generate the highest sales?
- Which product categories perform best?
- Are a few products dominating sales, or is demand evenly distributed?
- Which products perform consistently over time?

---

#### Sales Performance

- Which months generate the highest sales?
- Which cities contribute the most revenue?
- Are sales trends consistent throughout the year?
- Which customer segments contribute most during peak seasons?

### 2.1 What is a KPI?

A Key Performance Indicator (KPI) is a measurable business indicator used to evaluate how effectively an organization is achieving a specific business objective.

Unlike general metrics, KPIs are directly linked to business goals and help management monitor performance, identify problems, and make informed business decisions.

In this project, KPIs will be used to measure customer purchasing behaviour, product performance, and overall sales performance.

### 2.2 KPI vs Metric

| Metric | KPI |
|---------|-----|
| Measures any business activity. | Measures progress toward a specific business objective. |
| May or may not be important for decision-making. | Directly supports business decision-making. |
| Used for monitoring general performance. | Used for evaluating strategic business performance. |
| Example: Total Products | Example: Average Order Value |

### 2.3 Characteristics of a Good KPI

A good KPI should have the following characteristics:

- Directly aligned with a business objective.
- Measurable using available data.
- Actionable for business decision-making.
- Trackable over time to monitor trends.
- Easy to understand and communicate.

### 2.4 KPI Selection Framework

Each KPI selected for this project will be evaluated using the following framework:

1. Business Objective
   - Which business objective does the KPI support?

2. Business Rationale
   - Why is this KPI important for the business?

3. Formula
   - How is the KPI calculated?

4. Business Interpretation
   - What does a high or low value indicate?

5. Business Action
   - What decisions or actions can management take based on this KPI?

6. Limitations
   - What information does this KPI not provide?

### 4.5 Final Selected KPIs

Based on the defined business objectives and KPI selection framework, the following KPIs were selected for the dashboard:

1. Total Sales
2. Total Orders
3. Average Order Value (AOV)
4. Repeat Purchase Rate
5. Average Sales per Customer

These KPIs collectively measure overall sales performance, customer purchasing behaviour, and customer loyalty while supporting executive-level business decision-making.

## 3. KPI Implementation & Analysis

### 3.1 Environment Setup

The required Python libraries are imported to perform data manipulation, analysis, and visualization throughout this project.

In [2]:
# Data Manipulation
import pandas as pd
import numpy as np

# Data Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Notebook Settings
plt.style.use("default")
pd.set_option("display.max_columns", None)

print("Libraries imported successfully.")

Libraries imported successfully.


### 3.2 Load Dataset

The cleaned dataset prepared during Task 2 is loaded for KPI calculation, business analysis, and dashboard development.

In [3]:
# Load Dataset
df = pd.read_csv("../../task 02 - EDA and business analytics/cleaned_ecommerce_data.csv")

print("Dataset loaded successfully.")

Dataset loaded successfully.


In [4]:
# Convert Order_Date to datetime
df["Order_Date"] = pd.to_datetime(df["Order_Date"])

### 3.3 Quick Dataset Validation

Before calculating KPIs, a quick validation is performed to ensure that the dataset has been loaded correctly and contains the required information for analysis.

In [5]:
# Shape of dataset
print("Dataset Shape:", df.shape)

# Column Names
print("\nColumns:")
print(df.columns.tolist())

# Missing Values
print("\nMissing Values:")
print(df.isnull().sum())

# Data Types
print("\nData Types:")
print(df.dtypes)

Dataset Shape: (1000, 15)

Columns:
['Order_ID', 'Order_Date', 'Customer_ID', 'Customer_Name', 'Age', 'Gender', 'City', 'Product', 'Category', 'Quantity', 'Unit_Price', 'Total_Sales', 'Calculated_Sales', 'Month', 'Age_Group']

Missing Values:
Order_ID            0
Order_Date          0
Customer_ID         0
Customer_Name       0
Age                 0
Gender              0
City                0
Product             0
Category            0
Quantity            0
Unit_Price          0
Total_Sales         0
Calculated_Sales    0
Month               0
Age_Group           0
dtype: int64

Data Types:
Order_ID                       str
Order_Date          datetime64[us]
Customer_ID                    str
Customer_Name                  str
Age                        float64
Gender                         str
City                           str
Product                        str
Category                       str
Quantity                     int64
Unit_Price                 float64
Total_Sales     

### 4.1 KPI 1 - Total Sales

#### Business Purpose

Total Sales represents the overall revenue generated from all recorded transactions. It provides an executive-level summary of business performance and serves as the primary KPI for evaluating revenue generation.

#### Formula

**Total Sales = Sum of Total_Sales**

In [6]:
# KPI 1 : Total Sales

total_sales = df["Total_Sales"].sum()

print(f"Total Sales : ₹{total_sales:,.2f}")

Total Sales : ₹139,399,439.65


#### Business Interpretation

The dataset generated a total revenue of ₹139,399,439.65 across all recorded transactions.

However, this value alone is insufficient for business decision-making. Total Sales becomes meaningful only when compared with previous time periods, predefined business targets, seasonal trends, or industry benchmarks.

Therefore, this KPI serves as a high-level executive indicator and should be analyzed together with trend-based KPIs such as Monthly Sales Trend, Average Order Value, Total Orders, and Customer Growth.

#### Limitations

- Total Sales does not indicate whether business performance is improving or declining.
- It provides no information about profitability.
- It does not explain which products, customers, or regions contributed to the revenue.
- Trend analysis is required before making strategic decisions.

#### 5.4.1 Business Question

How does total sales vary across different months? Are there any seasonal trends, peak months, or low-performing periods that require business attention?

In [7]:
# Check available month values
print(df["Month"].unique())

<StringArray>
[ 'February',   'October',       'May',  'December',  'November',     'April',
     'March',    'August',      'July', 'September',      'June',   'January']
Length: 12, dtype: str


#### 4.2 Sales Trend Analysis

In [8]:
# Monthly Sales Trend
monthly_sales = (
    df.groupby("Month")["Total_Sales"]
      .sum()
      .reset_index()
)

monthly_sales

,Month,Total_Sales
0,April,12222700.17
1,August,9448471.22
2,December,12299904.44
3,February,11511181.46
4,January,10905580.72
5,July,11746226.80
6,June,12912332.64
7,March,13059899.94
8,May,10984689.25
9,November,12627620.22


In [9]:
# Extract Month Number
df["Month_Number"] = df["Order_Date"].dt.month
df[["Order_Date", "Month_Number"]].head()

,Order_Date,Month_Number
0,2025-02-25,2
1,2025-10-14,10
2,2025-05-13,5
3,2025-12-02,12
4,2025-11-20,11


In [10]:
# Define calendar month order
month_order = [
    "January", "February", "March", "April",
    "May", "June", "July", "August",
    "September", "October", "November", "December"
]

# Convert Month into ordered categorical data
df["Month"] = pd.Categorical(
    df["Month"],
    categories=month_order,
    ordered=True
)

In [11]:
monthly_sales = (
    df.groupby("Month")["Total_Sales"]
      .sum()
      .reset_index()
)

monthly_sales

,Month,Total_Sales
0,January,10905580.72
1,February,11511181.46
2,March,13059899.94
3,April,12222700.17
4,May,10984689.25
5,June,12912332.64
6,July,11746226.80
7,August,9448471.22
8,September,9179896.29
9,October,12500936.50


### 4.3 KPI 2 - Total Orders

##### Business purpose
Measures the number of unique orders placed by customers

##### Formula
Count of unique Order_ID

### 4.4 KPI 3 - Average Order Value (AOV)
#### Business Purpose
Measures the average revenue generated per order.

#### Formula
Total Sales / Total Orders

### 4.5 KPI 4 - Repeat Purchase Rate
#### Business Purpose
Measures the percentage of customers who placed more than one order. This helps evaluate customer retention and repeat buying behavior.

#### Formula
(Number of Customers with Repeat Purchases / Total Customers) * 100


### 4.6 KPI 5 - Average Sales per Customer
#### Business Purpose
Measures the average revenue generated per customer.

#### Formula
Total Sales / Total Customers

In [12]:
# =========================

# Executive KPI Calculations

# =========================

# Total Sales

total_sales = df['Total_Sales'].sum()

# Total Orders

total_orders = df['Order_ID'].nunique()

# Unique Customers

total_customers = df['Customer_ID'].nunique()

# Average Order Value (AOV)

average_order_value = total_sales / total_orders

# Average Sales per Customer

average_sales_per_customer = total_sales / total_customers

# Repeat Purchase Rate

customer_order_counts = df.groupby('Customer_ID')['Order_ID'].nunique()

repeat_customers = (customer_order_counts > 1).sum()

repeat_purchase_rate = (repeat_customers / total_customers) * 100

# Display Results

print(f'Total Sales: ₹{total_sales:,.2f}')

print(f'Total Orders: {total_orders:,}')

print(f'Average Order Value (AOV): ₹{average_order_value:,.2f}')

print(f'Average Sales per Customer: ₹{average_sales_per_customer:,.2f}')

print(f'Repeat Purchase Rate: {repeat_purchase_rate:.2f}%')

Total Sales: ₹139,399,439.65
Total Orders: 992
Average Order Value (AOV): ₹140,523.63
Average Sales per Customer: ₹147,201.10
Repeat Purchase Rate: 5.49%


### Business Interpretation

#### Total Sales
The business generated a total revenue of ₹139,399,439.65 from all recorded transactions.

#### Total Orders
 There are 992 unique orders, indicating the total transaction volume handled during the recorded period.

#### Average Order Value (AOV)
 Each order generated an average revenue of ₹140,523.63, suggesting that transactions in the dataset have relatively high monetary value.

#### Average Sales per Customer
 On average, each customer contributed ₹147,201.10 in revenue.

#### Repeat Purchase Rate
 Approximately 5.49% of customers placed more than one order, indicating that most customers in this dataset appear to be one-time buyers while a smaller group returned for additional purchases.

These KPIs provide a high-level executive summary of business performance and will serve as the primary metrics displayed on the interactive dashboard.

## 5.1 Customer Segmentation Analysis

Customer segmentation helps identify which customer groups contribute the most revenue and how purchasing behavior varies across different demographics. In this analysis, we will examine sales performance by Age Group and Gender to understand the characteristics of the customer base.

### Business Questions:

1. Which age group generates the highest sales?

2. How does sales contribution vary across age groups?

3. Which gender contributes more to total sales?

4. What insights can help the business target customer segments more effectively?

## 5.2 Segmentation Analysis Summary

For detailed exploratory analysis and visualizations, refer to Task 2 - EDA and Business Analytics. In this task, the focus is on summarizing key customer, product, and geographical segments that will be highlighted in the interactive dashboard.

The dashboard will include:

1. Sales by Age Group

2. Sales by Gender

3. Top Products

4. Category Performance

5. City-wise Sales Performance

6. Repeat Customer Insights

## 6. Deep-Dive Analysis: Customer Segmentation

This section performs a customer segmentation analysis using demographic attributes such as Age Group and Gender. The objective is to identify which customer segments contribute the highest revenue and derive actionable business recommendations.

In [14]:
# Sales by Age Group
age_sales = df.groupby('Age_Group')['Total_Sales'].sum().sort_values(ascending=False)
print(age_sales)

Age_Group
36-45    31533245.71
26-35    29632336.96
46-55    28216508.63
56-65    27733251.41
18-25    22284096.94
Name: Total_Sales, dtype: float64


### Key Findings: Age Group

1. The 36–45 age group generated the highest sales.

2. The 26–35 age group was the second-highest contributor.

3. The 18–25 age group generated the lowest sales.

4. Customers between 26 and 45 years are the primary revenue drivers.

In [15]:
# Sales by Gender
gender_sales = df.groupby('Gender')['Total_Sales'].sum().sort_values(ascending=False)
print(gender_sales)

Gender
Male      72463550.63
Female    66935889.02
Name: Total_Sales, dtype: float64


### Key Findings: Gender

1. Male customers generated slightly higher sales than female customers.

2. Both genders contribute significantly to the business.

3. Marketing campaigns should continue targeting both customer segments.

In [16]:
# Product-wise Sales
product_sales = df.groupby('Product')['Total_Sales'].sum().sort_values(ascending=False)
print(product_sales)

Product
Laptop    25443008.51
Mobile    25335573.19
Book      25031689.40
Rice      22231711.28
Chair     21521561.48
Shoes     19835895.79
Name: Total_Sales, dtype: float64


### Key Findings: Products

1. A small number of products contribute a large share of revenue.

2. High-performing products should receive inventory and promotional priority.

3. Lower-performing products may require pricing or marketing improvements.

In [17]:
# Category-wise Sales
category_sales = df.groupby('Category')['Total_Sales'].sum().sort_values(ascending=False)
print(category_sales)

Category
Electronics    50778581.70
Education      25031689.40
Grocery        22231711.28
Furniture      21521561.48
Fashion        19835895.79
Name: Total_Sales, dtype: float64


### Key Findings: Categories

1. Electronics is the highest revenue-generating category.

2. Other categories contribute comparatively lower sales.

3. Future promotional campaigns should prioritize top-performing categories.

In [18]:
# City-wise Sales
city_sales = df.groupby('City')['Total_Sales'].sum().sort_values(ascending=False)
print(city_sales)

City
Patna        19285966.89
Kolkata      18884349.57
Bengaluru    18773574.32
Mumbai       18757050.17
Hyderabad    17166766.87
Delhi        16097079.00
Pune         14513175.90
Gaya         14380859.39
Unknown       1540617.54
Name: Total_Sales, dtype: float64


### Key Findings: Cities

1. Patna generated the highest sales.

2. Sales performance varies across cities.

3. Geographic targeting can help allocate marketing budgets more effectively.

4. Unknown represents records where city information was missing and was filled during data cleaning.

## Business Recommendations

1. Focus marketing efforts on customers aged 26–45.

2. Increase promotions and inventory planning for Electronics.

3. Develop city-specific campaigns for high-performing locations.

4. Improve customer retention because the Repeat Purchase Rate (~5.49%) is relatively low.

5. Enhance data collection processes to reduce missing city information.

## Interactive Dashboard Development

As part of Task 3, an interactive dashboard was developed using Microsoft Power BI. The dashboard was designed to help stakeholders monitor business performance and explore customer, product, and geographic insights interactively.

### Dashboard Features

1. Executive Overview with key performance indicators.

2. Customer Segmentation Analysis using Age Group and Gender.

3. Product and Category Performance Analysis.

4. Geographic Sales Analysis by City.

5. Interactive Month Slicer for filtering dashboard results.

6. Multi-page navigation for different business perspectives.

### Dashboard Pages

1. Executive Overview: KPIs and monthly filtering.

2. Customer Segmentation: Age Group and Gender analysis.

3. Product Analysis: Product-wise and Category-wise sales performance.

4. Geographic Analysis: City-wise sales performance.

### Tools Used

1. Python (Pandas)

2. Jupyter Notebook

3. Microsoft Power BI

4. GitHub

The dashboard complements the notebook analysis by providing an interactive and executive-friendly view of the business insights discovered during the deep-dive analysis.

## Conclusion

This deep-dive customer segmentation analysis identified the most valuable customer groups, products, categories, and cities. The insights were further presented through an interactive Power BI dashboard containing Executive Overview, Customer Segmentation, Product Analysis, and Geographic Analysis pages. These findings can support better marketing, inventory planning, and customer retention strategies.